components/utils/useSessionGroups.ts

```diff
@@ -0,0 +1,181 @@
+        '9c81bbdb-8898-4259-afac-0dc73ff363b5',
+        '24ad37da-2c0b-4f5c-afde-3266217e6d80',
+        '3c2badde-1534-494b-a084-8ca5857d648d',
+        'f3a57e6c-0325-4898-bffd-2d3040f5dee9',
+        'adbcf783-1ab2-456b-ba41-1041f139e3f2',
+      ]),
+      timeEnd: conferenceDate.clone().set({ hour: 12, minute: 10 }),
+      timeStart: conferenceDate.clone().set({ hour: 11, minute: 50 }),
+    },
+    {
+      sessions: getSessionById(sessions, [
+        '9b7efb7a-64e0-41ac-9439-f65a662147da',
+        '6d6553c0-b678-434d-b94e-c46fe77c86eb',
+        '0bcae524-eb87-4080-b189-ab5c7d5ad5fa',
+        '5aba6e83-cfd9-4114-af80-f28de931d8c2',
+        'fd0518e0-a52c-44dd-84fb-61ce59c3cdb5',
+        '70537fd7-4e49-4100-97ee-ce79c71545d6',
+      ]),
+      timeEnd: conferenceDate.clone().set({ hour: 13, minute: 5 }),
+      timeStart: conferenceDate.clone().set({ hour: 12, minute: 20 }),
+    },
+    {
+      sessions: getSessionById(sessions, [
+        'a6eb8bb3-6086-4cb3-b024-d0a6c4dd3de3',
+        'a577e148-b1d7-42e1-a424-5d0db3107ae2',
+        '2fcea05c-96dc-4802-b8a9-14bcfee01a64',
+        '97792db7-0c73-4fee-91c3-00d7fe002540',
+        'fa861d2a-9597-4a98-8510-fc0dc0b400e6',
+        '83b6a640-935b-4e5e-b251-81c3d69c0129',
+      ]),
+      timeEnd: conferenceDate.clone().set({ hour: 14, minute: 25 }),
+      timeStart: conferenceDate.clone().set({ hour: 14, minute: 5 }),
+    },
+    {
+      sessions: getSessionById(sessions, [
+        'b2795175-d14d-4090-a62e-153d4534b916',
+        '80721e7b-b082-4b50-9a9d-136d3054b7b0',
+        '94a2f4b3-bd6e-4eb6-9917-baa3bcb3d41f',
+        'f548e402-d04d-4318-a8c6-d879b3f11d37',
+        '00311b92-6c21-47a8-b8d2-af325581d6f9',
+        '35e1174f-8d50-48db-a410-d53c3c8ddf73',
+      ]),
+      timeEnd: conferenceDate.clone().set({ hour: 15, minute: 20 }),
+      timeStart: conferenceDate.clone().set({ hour: 14, minute: 35 }),
+    },
+    {
+      sessions: getSessionById(sessions, ['4c019f6f-c312-4bb9-8024-3352f6034d6e']),
+      timeEnd: conferenceDate.clone().set({ hour: 17, minute: 5 }),
+      timeStart: conferenceDate.clone().set({ hour: 16, minute: 20 }),
+    },
+  ]
+}
+
+// so manual - ideally there would be a better way to achieve this or expand it to handle the agenda too
+// e.g. on the agenda page show next sessions up top then the whole list
+export function useSessionGroups(
+  conferenceDate: Moment,
+  conferenceEndDate: Moment,
+  sessions: Session[],
+): SessionGroups {
+  const allSessionGroups: SessionGroup[] = React.useMemo(() => createSessionGroups(conferenceDate, sessions), [
+    conferenceDate.toString(),
+  ])
+  const isConferenceDay = dateTimeProvider.now().Value.isBetween(conferenceDate, conferenceEndDate)
+
+  let pastSessionsIndex: number = -1
+  let previousSessionIndex: number = -1
+  let nextSessionIndex: number = -1
+  let currentSessionIndex: number = -1
+  let futureSessionIndex: number = -1
+
+  for (let i = 0; i < allSessionGroups.length; i++) {
```

Comment:
I tried several variations with this. Grabbing the next session, mapping, reducer and all had their benefits/drawbacks. I don't know if this is the best approach or not but it's mostly clean

Reply:
Yeah it's not the prettiest, but if it works and works reliably then I'm happy